synapse/handlers/profile.py

```diff
@@ -218,6 +225,13 @@ def set_avatar_url(self, target_user, requester, new_avatar_url, by_admin=False)
         if not by_admin and target_user != requester.user:
             raise AuthError(400, "Cannot set another user's avatar_url")
 
+        if not by_admin and self.hs.config.disable_set_avatar_url:
+            profile = yield self.store.get_profileinfo(target_user.localpart)
+            if profile.avatar_url:
+                raise SynapseError(
+                    400, "Changing avatar url is disabled on this server"
```

Comment:
Same as above, slight rewording to make the error easier to understand for the user:

```suggestion
                    400, "Changing avatar is disabled on this server"
```